src/AlertsManagement/AlertsManagement/AlertCommands/UpdateAzureAlertState.cs

```diff
@@ -0,0 +1,61 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Rest.Azure;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.AlertsManagement.OutputModels;
+using Microsoft.Azure.Management.AlertsManagement.Models;
+
+namespace Microsoft.Azure.Commands.AlertsManagement
+{
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "AlertState", SupportsShouldProcess = true)]
+    [OutputType(typeof(PSAlert))]
+    public class UpdateAzureAlertState : AlertsManagementBaseCmdlet
+    {
+        #region Parameters declarations
+        /// <summary>
+        /// Alert Id
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   HelpMessage = "Unique Identifier of Alert / ResourceId of alert.")]
+        [ValidateNotNullOrEmpty]
+        [Alias("ResourceId")]
+        public string AlertId { get; set; }
```

Comment:
This should also take an alert tule object, and alert rule identity propetrties (rgname, name, etc.)

Reply:
Alert Id is a Guid and uniquely identifies an alert instance which needs to be updated. Alert rule identity properties altogether do not point to a unique instance. What is the intention here?

Reply:
I am asking for the required InputObject parameter set for this cmdlet, as agreed in our design review.

Reply:
Committed.